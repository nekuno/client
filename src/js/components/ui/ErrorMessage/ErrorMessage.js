import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './ErrorMessage.scss';

export default class ErrorMessage extends Component {

    constructor(props) {
        super(props);

        this.state = {
            display: 'block'
        };
    }

    static propTypes = {
        text: PropTypes.string.isRequired
    };

    handleClickCancel() {
        this.setState({display: 'none'});
    }

    render() {
        const {text} = this.props;
        const {display} = this.state;

        return (
            <div className={styles.errorMessageWrapper} style={{display: display}}>
                <div className={'small ' + styles.errorMessage}>
                    <div className={styles.message}>{text}</div>
                    <div className={'icon icon-x ' + styles.crossIcon} onClick={this.handleClickCancel.bind(this)}/>
                </div>
            </div>
        );
    }
}