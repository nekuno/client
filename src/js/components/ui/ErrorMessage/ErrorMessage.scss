@import '../../../../scss/_partials/colors';

.error-message-wrapper {

  .error-message {
    background: $red-error;
    box-sizing: border-box;
    color: $white-background;
    height: 48px;
    position: absolute;
    bottom: 0;
    width: 100%;

    .message {
      display: inline-block;
      line-height: 48px;
      width: 80%;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      padding: 0 5%;
      position: relative;
      -webkit-box-flex: 1;
      -webkit-flex-shrink: 1;
      -ms-flex: 0 1 auto;
      flex-shrink: 1;
      min-width: 0;
    }
    .cross-icon {
      display: inline-block;
      line-height: 48px;
      height: 48px;
      overflow: hidden;
      width: 10%;
    }
  }
}