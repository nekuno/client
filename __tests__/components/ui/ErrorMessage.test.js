import React from 'react';
import ErrorMessage from '../../../src/js/components/ui/ErrorMessage/ErrorMessage.js';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test ErrorMessage component', () => {
    const errorMessage = shallow(<ErrorMessage text={'Lorem ipsum'}/>);

    it('should be defined', () => {
        expect(ErrorMessage).toBeDefined();
    });
    it('should render correctly', () => {
        expect(errorMessage).toMatchSnapshot();
    });
});
