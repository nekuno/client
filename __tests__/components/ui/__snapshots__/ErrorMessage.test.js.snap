// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ErrorMessage component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorMessage
    text="Lorem ipsum"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="small errorMessage"
      >
        <div
          className="message"
        >
          Lorem ipsum
        </div>
        <div
          className="icon icon-x crossIcon"
          onClick={[Function]}
        />
      </div>,
      "className": "errorMessageWrapper",
      "style": Object {
        "display": "block",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="message"
          >
            Lorem ipsum
          </div>,
          <div
            className="icon icon-x crossIcon"
            onClick={[Function]}
          />,
        ],
        "className": "small errorMessage",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem ipsum",
            "className": "message",
          },
          "ref": null,
          "rendered": "Lorem ipsum",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "icon icon-x crossIcon",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="small errorMessage"
        >
          <div
            className="message"
          >
            Lorem ipsum
          </div>
          <div
            className="icon icon-x crossIcon"
            onClick={[Function]}
          />
        </div>,
        "className": "errorMessageWrapper",
        "style": Object {
          "display": "block",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="message"
            >
              Lorem ipsum
            </div>,
            <div
              className="icon icon-x crossIcon"
              onClick={[Function]}
            />,
          ],
          "className": "small errorMessage",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lorem ipsum",
              "className": "message",
            },
            "ref": null,
            "rendered": "Lorem ipsum",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "icon icon-x crossIcon",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
